fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java

```diff
@@ -0,0 +1,222 @@
+                            " must has alias except base select item");
+                }
+            }
+            // analyze query statement, can check table and column is exists in meta
+            Analyzer.analyze(queryStatement, context);
+            // check table is in this database and is OlapTable
+            Map<String, TableRelation> tableRelationHashMap = new HashMap<>();
+            extractTableRelation(selectRelation.getRelation(), tableRelationHashMap);
+            tableRelationHashMap.forEach((tbl, tableRelation) -> {
+                if (!tableRelation.getName().getDb().equals(statement.getTableName().getDb())) {
+                    throw new SemanticException("Materialized view not support table which in other database");
+                }
+                if (!(tableRelation.getTable() instanceof OlapTable)) {
+                    throw new SemanticException("Materialized view only support olap tables");
+                }
+            });
+            // set column
+            setColumn(statement, selectRelation);
+            // check partition SlotRef
+            checkExpInColumn(partitionExpDesc, statement);
+            // check partition Expr
+            checkPartitionExpParams(partitionExpDesc);
+            //check partition key must be base table partition key
+            checkPartitionKey(partitionExpDesc, tableRelationHashMap);
+            // check distribution
+            checkDistribution(statement);
+            //set view def
+            String viewSql = ViewDefBuilder.build(queryStatement);
+            statement.setInlineViewDef(viewSql);
+            return null;
+        }
+
+        private void setColumn(CreateMaterializedViewStatement statement, QueryRelation queryRelation) {
+            List<Column> mvColumns = Lists.newArrayList();
+            Map<Column, Expr> columnExprMap = Maps.newHashMap();
+            List<String> columnOutputNames = queryRelation.getColumnOutputNames();
+            List<Expr> outputExpression = queryRelation.getOutputExpression();
+            for (int i = 0; i < outputExpression.size(); ++i) {
+                Column column = new Column(columnOutputNames.get(i), outputExpression.get(i).getType());
+                mvColumns.add(column);
+                columnExprMap.put(column, outputExpression.get(i));
+            }
+            statement.setMvColumnItems(mvColumns);
+            statement.setColumnExprMap(columnExprMap);
+        }
+
+        private void checkExpInColumn(PartitionExpDesc partitionExpDesc, CreateMaterializedViewStatement statement) {
+            boolean isInColumn = false;
+            List<Column> mvColumnItems = statement.getMvColumnItems();
+            Map<Column, Expr> columnExprMap = statement.getColumnExprMap();
+            for (Column mvColumnItem : mvColumnItems) {
+                if (partitionExpDesc.getSlotRef().getColumnName().equals(mvColumnItem.getName())) {
```

Comment:
```suggestion
                if (partitionExpDesc.getSlotRef().getColumnName().equalsIgnoreCase(mvColumnItem.getName())) {
```
column name is case insensitive